trestle/core/commands/add.py

```diff
@@ -36,93 +32,63 @@
 logger = logging.getLogger(__name__)
 
 
-class AddCmd(CommandPlusDocs):
-    """Add an OSCAL object to the provided file based on element path.
-
-    This CLI takes input a filename and a list of comma-seperated element path. Element paths are based on the json
-    field names.
+class Add():
+    """
+    Class supporting Add of an OSCAL object to a provided file based on element path.
 
     Examples of element paths:
         catalog.metadata
         catalog.controls.control
         assessment-results.results.
 
-
     The method first finds the parent model from the file and loads the file into the model.
     Then the method executes 'add' for each of the element paths specified.
+    Add was originally its own command but has been incorporated into the Create command.
     """
 
-    name = 'add'
+    def add_from_args(self, args: argparse.Namespace) -> int:
+        """Parse args for add element to file."""
+        file_path = pathlib.Path(args.file).resolve()
 
-    def _init_arguments(self) -> None:
-        self.add_argument(
-            f'-{const.ARG_FILE_SHORT}',
-            f'--{const.ARG_FILE}',
-            help=const.ARG_DESC_FILE + ' to add component/subcomponent to.',
-            required=True
-        )
-        self.add_argument(
-            f'-{const.ARG_ELEMENT_SHORT}',
-            f'--{const.ARG_ELEMENT}',
-            help=const.ARG_DESC_ELEMENT + ' to add.',
-            required=True
-        )
-        self.add_argument(const.IOF_SHORT, const.IOF_LONG, help=const.IOF_HELP, action='store_true')
+        # Get parent model and then load json into parent model
+        parent_model, _ = ModelUtils.get_stripped_model_type(file_path, args.trestle_root)
+        parent_object = parent_model.oscal_read(file_path)
+        # FIXME : handle YAML files after detecting file type
+        parent_element = Element(parent_object, classname_to_alias(parent_model.__name__, AliasMode.JSON))
 
```

Comment:
Does code for YAML needs to be added? In documentation we say all our commands work with YAML.

Reply:
This is an old comment and doesn't apply.  The code already handles json and yaml and I have added more comment lines to explain.  The AliasMode.json here is special and isn't related to the output file type.